src/coreclr/src/gc/gc.cpp

```diff
@@ -11506,6 +11507,12 @@ BOOL gc_heap::grow_heap_segment (heap_segment* seg, uint8_t* high_address, bool*
 
         assert (heap_segment_committed (seg) <= heap_segment_reserved (seg));
         assert (high_address <= heap_segment_committed (seg));
+
+#ifdef MULTIPLE_HEAPS
+        assert (!gradual_decommit_in_progress_p ||
+                 heap_segment_decommit_target (seg) == nullptr ||
+                 heap_segment_committed (seg) <= heap_segment_decommit_target (seg));
+#endif // MULTIPLE_HEAPS
```

Comment:
I must be missing something - isn't decommit_target smaller than committed? we keep shrinking committed till we reach decommit_tagert when we do the decommit steps. 

also when gradual_decommit_in_progress_p is TRUE, shouldn't decommit_target always be non NULL? 

nit -
```
        assert (!gradual_decommit_in_progress_p ||
                 (heap_segment_decommit_target (seg) == nullptr) ||
                 (heap_segment_committed (seg) <= heap_segment_decommit_target (seg)));
```

Reply:
For some heaps, decommit_target may be bigger than committed, and that's fine - it just means that we won't decommit.

When gradual_decommit_in_progress_p is TRUE, decommit_target will be non NULL for the ephemeral_heap_segment, but may be NULL for other segments - the condition was meant to filter out the other segments. Now that I think about it, I realize this isn't a good test - I should compare against ephemeral_heap_segment directly. So I fixed the assert as follows:

        // we should never increase committed beyond decommit target when gradual
        // decommit is in progress - if we do, this means commit and decommit are
        // going on at the same time.
        assert (!gradual_decommit_in_progress_p ||
                (seg != ephemeral_heap_segment) ||
                (heap_segment_committed (seg) <= heap_segment_decommit_target (seg)));

Reply:
oh right, I was only thinking about eph seg. and yes the point is if we actually need to commit more, the decommit target had better be bigger otherwise we have a problem. thanks! I agree the new assert looks better.